### PR TITLE
MM-37678: Disallow title editing if user has no permissions to change channel name

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
While we decide how to handle run titles decoupled from channel names, this PR completely disables the ability to edit the title from the RHS when the user has no permissions to change the channel name.

This mimics the feature to change the channel name from the channel header: the menu entry `Rename Channel` is only shown to users that have permissions to actually change it.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37678

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

**Comment by agarciamontoro** [comment]
@abhijit-singh, I think this makes sense from a UX standpoint, but assigning you here to double-check.

**Review by lieut-data — APPROVED** [review]
Classy!

**Review by abhijit-singh — APPROVED** [review]

File: webapp/src/components/rhs/rhs_about_title.tsx [hunk 1]
```diff
@@ -74,6 +89,18 @@ const RHSAboutTitle = (props: Props) => {
...
+    const channel = getCurrentChannel(state);
+
+    const permission = channel.type === General.OPEN_CHANNEL ? Permissions.MANAGE_PUBLIC_CHANNEL_PROPERTIES : Permissions.MANAGE_PRIVATE_CHANNEL_PROPERTIES;
+
+    return haveIChannelPermission(state, {
+        channel: channel.id,
+        team: channel.team_id,
+        permission,
+    });
+};
```
Comment (cpoile) [inline webapp/src/components/rhs/rhs_about_title.tsx:102]: I'm wondering... Have you viewed the react profiler after this change? Will this cause the whole component to re-render on any store change?

**Review by cpoile — CHANGES_REQUESTED** [review]
Just a quick question, otherwise nice!

File: webapp/src/components/rhs/rhs_about_title.tsx (hunk 1 above)
Comment (agarciamontoro) [reply webapp/src/components/rhs/rhs_about_title.tsx:102]: I have not. Taking a look, thanks!

Comment (agarciamontoro) [reply webapp/src/components/rhs/rhs_about_title.tsx:102]: @cpoile, I'm checking with the paint flashing tool and making changes in the store does not make the component re-render. Maybe I'm not testing it as expected, would you have done something different to check your concern?

Comment (cpoile) [reply webapp/src/components/rhs/rhs_about_title.tsx:102]: Nope, that's perfect, thanks @agarciamontoro !

**Review by cpoile — APPROVED** [review]
Nice!

**Pull request merged**